### ISSUE TITLE
Package Installers

### ISSUE THREAD

**Issue by oliver-leung** [AguaClara/aguaclara_tutorial#22]
We should make an installer package to automatically install as many of these programs as possible. I have a rudimentary Windows batch file that runs all of the necessary installs, and I'm thinking of making a shell script for Unix (Mac OS and Linux) systems.

**Comment by oliver-leung** [comment]
A Windows installer has been created, I will zip it and upload it to the AguaClara Google Drive.

**Comment by eak24** [comment]
@oliver-leung what is this installer for installing? Robust installers are notoriously difficult to write.

**Comment by oliver-leung** [comment]
@ethan92429 This is for installing Atom, Anaconda/Python, and Git and setting them all up. It doesn't actually "next" through each of the installers, but it opens them and then runs complementary setup where necessary:
* Installs the Sync Settings package and configures the Gist ID of settings
* pip installs aide_design and aguaclara_research
* Unsets the git credentials helper for shared computers

I started this out because I was given the task of installing these programs on all of the lab computers in Hollister B60, so I created a batch file for automating the process. I thought it would be a great idea for compiling these installers for AguaClara team members to set up their own computers, if they wanted to.

**Comment by eak24** [comment]
Cool! That sounds like a great application of a bash script! especially when all the computers you are installing to are configured the same way. I recommend putting this instead in the aguaclara_tutorial repo.

**Comment by oliver-leung** [comment]
I tried uploading the entire installer to the repo, but it took a ridiculously long time to get the .exe install files to upload because they're quite large, even when zipped. If we ever need to change them, it'd be a hassle to do so. Instead, in the wiki, I can put a link to a zip file on the AguaClara Google Drive folder.

**Comment by eak24** [comment]
Instead of downloading the whole program (or multiple), you could just use [BITSAdmin](https://superuser.com/questions/299754/wget-curl-alternative-native-to-windows) to download it using the computer's own tools. That way, to change the version downloaded, you change one number in the URL rather than upload a new zip file.  @oliver-leung

**Comment by oliver-leung** [comment]
That's quite a useful tool - I'll try to implement it!

The reason why I packaged all of the .exe installers with the script is because I wanted to be able to download everything to a flash drive, plug it into a lab computer (or someone else's computer) and just install from that. This would avoid leaving any extra residual files on the computer - I suppose the same thing could be done with a BITSAdmin implementation via `rm`.

Also, Sync Settings doesn't install from a .exe, but rather by downloading a folder to the user's .atom folder. Right now, I have a copy of the Sync Settings folder in the installer, which is then copied to .atom.

I just realized that we can run `apm install sync-settings` instead! That might cause some issues if `apm` isn't added to path after the Atom install, but I can test that and see.
